iothub/service/src/Resources/Fx.cs

```diff
@@ -64,7 +64,7 @@ public static void AssertAndThrow(bool condition, string description)
         public static Exception AssertAndThrow(string description)
         {
             Assert(description);
-            throw Exception.AsError(new AssertionFailedException(description));
+            throw Exception.AsError(new FatalException(description));
```

Comment:
What does this change mean?

Reply:
I'm not completely sure, but we don't use this AssertionFailedException type anywhere else and we do use this FatalException type elsewhere in this class. I'd like to delete the AssertionFailedException class since it only has this one use.

Reply:
Will it impact customers?

Reply:
Not at all. The only place this function is called is by the Message class if the user attempts to clone a message that uses a Stream for its body. As I understand, that use case will disappear when we remove streams from Message classes, so this function won't even be called anymore.